std/functional.d

```diff
@@ -104,31 +104,7 @@ the parameter or provide the symbol via the $(D parmName) argument.
 If $(D fun) is not a string, $(D unaryFun) aliases itself away to $(D fun).
 */
 
-template unaryFun(alias fun, string parmName = "a")
-{
-    static if (is(typeof(fun) : string))
-    {
-        static if (!fun._ctfeMatchUnary(parmName))
-        {
-            import std.traits, std.typecons, std.meta;
-            import std.algorithm, std.conv, std.exception, std.math, std.range, std.string;
-        }
-        auto unaryFun(ElementType)(auto ref ElementType __a)
-        {
-            mixin("alias " ~ parmName ~ " = __a ;");
-            return mixin(fun);
-        }
-    }
-    else static if (needOpCallAlias!fun)
-    {
-        // Issue 9906
-        alias unaryFun = fun.opCall;
-    }
-    else
-    {
-        alias unaryFun = fun;
-    }
-}
+alias unaryFun(alias fun, string parmName = "a") = naryFun!(fun, parmName);
```

Comment:
Here's what it looks like as a template with [external enums](http://dpaste.dzfl.pl/69f1a42d30a1] and [nested enums](http://dpaste.dzfl.pl/250bb6a39c64). The `pragma` is triggered each time for the latter case.